src/main/java/uk/ac/cam/cl/dtg/segue/database/PostgresSqlDb.java

```diff
@@ -86,4 +86,18 @@ public void close() {
         }
 
     }
+
+    /**
+     * Check whether the database is a read only replica or not.
+     * @return whether the database is read only.
+     */
+    public boolean isReadOnlyReplica() throws SQLException {
+        try (Connection conn = getDatabaseConnection();
+             PreparedStatement pst = conn.prepareStatement("SELECT pg_is_in_recovery()");
+             ResultSet results = pst.executeQuery();
+        ) {
```

Comment:
Nice! I was going to have a go at fixing that LGTM error. I think it should apply to almost all our database code!

Reply:
This felt like a good test-case, since the statement contains no variables and there's nothing in the try beyond the expected auto-closeable resources. If LGTM doesn't comment below this commit, then success and a potential model going forward.